src/mixins/raster-layer-poly-mixin.js

```diff
@@ -565,17 +494,153 @@ export default function rasterLayerPolyMixin(_layer) {
       }
     }
 
-    const defaultMarkOptions = {
-      strokeColor: "white",
-      lineJoin: "miter",
-      miterLimit: 10,
-      strokeWidth: 0
+    return {
+      scale,
+      fillColor
+    }
+  }
+
+  _layer._updateFromMetadata = (metadata, layerName = "") => {
+    if (usesAutoColors() && Array.isArray(metadata.scales)) {
+      const colorScaleName = getColorScaleName(layerName)
+      for (const scale of metadata.scales) {
+        if (scale.name === colorScaleName) {
+          _layer.colorDomain(scale.domain)
+        }
+      }
     }
-    const mark =
-      typeof state.mark === "object" ? state.mark : defaultMarkOptions
+  }
 
-    const marks = [
-      {
+  _layer.__genVega = function({
+    chart,
+    bboxFilter,
+    filter,
+    globalFilter,
+    layerFilter = [],
+    lastFilteredSize,
+    filtersInverse,
+    layerName,
+    useProjection
+  }) {
+    const autocolors = usesAutoColors()
+    const getStatsLayerName = () => layerName + "_stats"
+    const state = _layer.getState()
+
+    let data;
+    if (isContourType(state)) {
```

Comment:
Very dramatic diff for some reason, this check was what I added `if (isContourType(state))`